test/optim/test_optim.py

```diff
@@ -736,7 +736,7 @@ def _test_derived_optimizers_varying_tensors(self, optimizer_with_kwargs, kwarg)
                     actual = mt_p_state[k]
                     self.assertEqual(st_p_state[k], actual, rtol=rtol, atol=atol)
 
-    def _test_derived_optimizers(self, optimizer_pairs_with_flags, flag):
+    def _test_derived_optimizers(self, optimizer_pairs_with_flags, flag, reduced_precision=False):
```

Comment:
Don't think this tests sparseadam. I think it would have previously failed too.

Reply:
Sure, I recall this is actually a high prio as well. Let me migrate `Adadelta, RMSProp and Rprop` next.